client/modules/IDE/components/FileNode.jsx

```diff
@@ -67,6 +67,7 @@ export class FileNode extends React.Component {
   validateFileName() {
     const oldFileExtension = this.originalFileName.match(/\.[0-9a-z]+$/i);
     const newFileExtension = this.props.name.match(/\.[0-9a-z]+$/i);
+    const fileName = this.props.name.replace(/\.[^/.]+$/, '');
```

Comment:
what does this regular expression do?

Reply:
@catarak It gets the filename **without** the extension

Reply:
got it! could you name this variable "fileNameWithoutExtension" so it's clear?

Reply:
@catarak Sure. I've fixed it.